app/controllers/authors_controller.rb

```diff
@@ -0,0 +1,42 @@
+class AuthorsController < ApplicationController
+
+  def show
+    @author = Author.find(params[:id])
+  end
+
+  def index
+    @authors = Author.all
+  end
+
+  def new
+    @author = Author.new
+  end
+
+  def create
+    @author = Author.new(params.require(:author).permit(:first_name, :socond_name, :date_of_birth, :wikipedia_url, :goodreads_url))
+    @author.user = current_user
+    if @author.save
+      flash[:notice] = "Author was saved"
+      redirect_to authors_path
+    else
+      render 'new'
+    end
+  end
+
+
+  def edit
+  end
+
+  def update
+    if @author.update(params.require(:author).permit(:first_name, :socond_name, :date_of_birth, :wikipedia_url, :goodreads_url))
```

Comment:
...and here - the same piece of code.
If you need you change it, you will always need to change it in 2 places, which is not convenient.
let't add a private method `author_params` and put this:
`params.require(:author).permit(:first_name, :socond_name, :date_of_birth, :wikipedia_url, :goodreads_url)`

into the method. then you'll be able to use the method in these 2 places ;)

Reply:
does it even works? what is @author here? where is it defined?